tasks/main.yml

```diff
@@ -0,0 +1,55 @@
+---
+- name: Set version specific variables
+  include_vars: "{{ item }}"
+  with_first_found:
+    - "{{ ansible_distribution }}-{{ ansible_distribution_version }}.yml"
+    - "{{ ansible_distribution }}-{{ ansible_distribution_major_version }}.yml"
+    - "{{ ansible_distribution }}.yml"
+    - "{{ ansible_os_family }}.yml"
+    - "default.yml"
+
+- name: Ensure certificate role dependencies are installed
+  package:
+    name: "{{ __certificate_packages }}"
+    state: present
+
+- name: Find required providers
+  set_fact:
+    __certificate_providers: |
+      {{
+        certificate_requests |
+        map(attribute='provider', default=__certificate_provider_default) |
+        unique |
+        list
+      }}
+
+- name: Ensure provider packages are installed
+  package:
+    name: "{{ __certificate_provider_vars[__certificate_provider].packages }}"
+    state: present
+  loop: "{{ __certificate_providers }}"
+  loop_control:
+    loop_var: __certificate_provider
+  when: |
+    __certificate_provider_vars[__certificate_provider].packages is defined
+
+# While most Linux system roles have a list of services
+#   this role doesn't need it. In fact some providers
+#   won't have any service at all.
+- name: Ensure provider service is running
+  service:
+    name: "{{ __certificate_provider_vars[__certificate_provider].service }}"
+    state: started
+    enabled: yes
+  loop: "{{ __certificate_providers }}"
+  loop_control:
+    loop_var: __certificate_provider
+  when: "__certificate_provider_vars[__certificate_provider].service is defined"
+
+- name: Ensure certificate requests
+  certificate_request:
+    name: "{{ item.name | default(omit) }}"
+    dns: "{{ item.dns | default(omit) }}"
+    ca: "{{ item.ca | default(omit) }}"
+    provider: "{{ item.provider | default(omit) }}"
+  loop: "{{ certificate_requests }}"
```

Comment:
Sorry for this novice question. Having default(omit) on all the dict keys, certificate_request accepts, e.g., a request missing name but having the rest?

Reply:
Since `certificate_request` is a python module the module implementation is taking care of mandatory/not mandatory and also the defaults. Omitting was just a way to delegate this responsibility to the module.

Reply:
I see. I.e., I can request a cert with none of the values?

Reply:
No. If you don't set any value it will raise an error saying that `name` is required. If you set name if should complain that you didn't set `ca`. If you set `ca` I'm not sure what would happen (I'll test to check).

`ca` and `name` are declared as mandatory in the file `library/certificate_request.py`, more specifically in https://github.com/seocam/certificate-system-role/blob/49aa94bdbd789e0885a2b1620d74b82c95c91b1b/library/certificate_request.py#L93-L100

Reply:
@nhosoi it's indeed possible to issue a certificate without setting the dns and it should work (certmonger allows it) but the module code was not working. 

This is the commit with the fix: 875e8c2ef9142105897cba78af13dfd223eff504

Reply:
Thanks for the clarification, @seocam!